util/helm/client.go

```diff
@@ -192,11 +196,13 @@ func (c *nativeHelmChart) ExtractChart(chart string, version string, project str
 			// 'helm pull' ensures that chart is downloaded into temp directory
 			_, err = helmCmd.PullOCI(c.repoURL, chart, version, tempDest, c.creds)
 			if err != nil {
+				_ = os.RemoveAll(tempDir)
 				return "", nil, err
 			}
 		} else {
 			_, err = helmCmd.Fetch(c.repoURL, chart, version, tempDest, c.creds, passCredentials)
 			if err != nil {
+				_ = os.RemoveAll(tempDir)
```

Comment:
```suggestion
				_ = os.RemoveAll(tempDest)
```

Reply:
@nitishfy this is exactly the issue - there are 2 different directories created - one for fetching purposes (tempDest) and another one at the beginning of the function for extraction. tempDest is removed using a defer function. tempDir on the other hand is supposed to be removed by the closer - after manifest generation is done, but in case of error the closer is never called so this code handles this case. See https://github.com/argoproj/argo-cd/issues/16994#issuecomment-1964345641

Reply:
thanks for confirming!